rust/operator-binary/src/spark_k8s_controller.rs

```diff
@@ -452,12 +445,14 @@ fn pod_template(
     env: &[EnvVar],
     s3conn: &Option<S3ConnectionSpec>,
     s3logdir: &Option<S3LogDir>,
+    spark_image: &ResolvedProductImage,
 ) -> Result<PodTemplateSpec> {
     let container_name = SparkContainer::Spark.to_string();
     let mut cb = ContainerBuilder::new(&container_name).context(IllegalContainerNameSnafu)?;
     cb.add_volume_mounts(config.volume_mounts.clone())
         .add_env_vars(env.to_vec())
-        .resources(config.resources.clone().into());
+        .resources(config.resources.clone().into())
+        .image_pull_policy(&spark_image.image_pull_policy);
```

Comment:
Theres a method `image_from_product_image` in the container builder that sets the image and pull policy.